salt/grains/core.py

```diff
@@ -2746,3 +2746,37 @@ def default_gateway():
         except Exception:
             continue
     return grains
+
+
+def _kernelparams(cmdline):
+    '''
+    Return the kernel boot parameters
+    '''
+
+    try:
+        import shlex
+    except ImportError:
```

Comment:
I will try to use `salt.utils.shlex_split` instead but split using whitespace is not enough because kernel parameter could be like below:
```
param="value1 value"
```